File: examples/server/clients/test_websocket.js
#!/usr/bin/env node

const WebSocket = require('ws');

// Test WebSocket connection to whisper server
const ws = new WebSocket('ws://127.0.0.1:8081');

ws.on('open', function open() {
    console.log('✅ Connected to whisper WebSocket server');
    
    // Test session start
    console.log('📤 Sending audio.start message...');
    ws.send(JSON.stringify({
        type: 'audio.start'
    }));
});

ws.on('message', function message(data) {
    try {
        const msg = JSON.parse(data.toString());
        console.log('📥 Received message:', msg);
        
        if (msg.type === 'session.start') {
            console.log(`🎯 Session started: ${msg.session_id}`);
        } else if (msg.type === 'audio.started') {
            console.log('🎵 Audio processing started');
            
            // Send audio end to complete the test
            setTimeout(() => {
                console.log('📤 Sending audio.end message...');
                ws.send(JSON.stringify({
                    type: 'audio.end'
                }));
            }, 1000);
        } else if (msg.type === 'audio.ended') {
            console.log('✅ Audio processing completed');
            console.log('🎉 WebSocket test successful!');
            ws.close();
        }
    } catch (e) {
        console.log('📥 Raw message:', data.toString());
    }
});

ws.on('error', function error(err) {
    console.error('❌ WebSocket error:', err.message);
});

ws.on('close', function close() {
    console.log('🔌 WebSocket connection closed');
});

// Timeout after 10 seconds
setTimeout(() => {
    console.log('⏰ Test timeout');
    process.exit(1);
}, 10000);

File: examples/server/clients/test_websocket.py
#!/usr/bin/env python3
"""
Simple WebSocket test client for whisper.cpp server
"""
import asyncio
import websockets
import json
import sys

async def test_websocket():
    uri = "ws://127.0.0.1:8081"
    
    try:
        print("🔌 Connecting to whisper WebSocket server...")
        async with websockets.connect(uri) as websocket:
            print("✅ Connected successfully!")
            
            # Listen for welcome message
            response = await websocket.recv()
            msg = json.loads(response)
            print(f"📥 Received: {msg}")
            
            if msg.get('type') == 'session.start':
                print(f"🎯 Session started: {msg.get('session_id')}")
                
                # Send audio.start
                print("📤 Sending audio.start...")
                await websocket.send(json.dumps({"type": "audio.start"}))
                
                # Wait for response
                response = await websocket.recv()
                msg = json.loads(response)
                print(f"📥 Received: {msg}")
                
                if msg.get('type') == 'audio.started':
                    print("🎵 Audio processing started")
                    
                    # Send audio.end
                    print("📤 Sending audio.end...")
                    await websocket.send(json.dumps({"type": "audio.end"}))
                    
                    # Wait for response
                    response = await websocket.recv()
                    msg = json.loads(response)
                    print(f"📥 Received: {msg}")
                    
                    if msg.get('type') == 'audio.ended':
                        print("✅ Audio processing completed")
                        print("🎉 WebSocket test successful!")
                        return True
            
            return False
            
    except Exception as e:
        print(f"❌ WebSocket error: {e}")
        return False

if __name__ == "__main__":
    result = asyncio.run(test_websocket())
    sys.exit(0 if result else 1)
